### ISSUE TITLE
ChainDB: ChainDB should be able to return chain fragments with associated ledger states

### ISSUE THREAD

**Issue by edsko** [input-output-hk/ouroboros-network#1440]
This will be required in order to be able to create a fork in block production; see #1437 for details. The tentative API proposal is

```haskell
getLedgerHistory :: STM m (Lazy.AnchoredFragment (Header blk, ExtLedgerState blk))
```

**Comment by edsko** [comment]
This will be required for the implementation of the local state query protocol also (#1366).

**Comment by edsko** [comment]
Due to concerns about laziness, will implement

```haskell
    , getPastLedger      :: STM m (Point -> Maybe (ExtLedgerState blk))
```

instead.

**Comment by edsko** [comment]
Turns out, this is not implementable (either function). The problem is that computing an arbitrary ledger state may involve reading from disk, and hence this cannot live in STM. We need to split up block production in order to do this; that is, we need to close https://github.com/input-output-hk/ouroboros-network/issues/786 first.

**Comment by edsko** [comment]
Marking as high priority because required to fix a bug (#1437).
